hypershift-operator/controllers/hostedcluster/internal/platform/openstack/openstack.go

```diff
@@ -66,9 +69,66 @@ func (a OpenStack) ReconcileCAPIInfraCR(ctx context.Context, client client.Clien
 	}); err != nil {
 		return nil, err
 	}
+
+	err := reconcileOpenStackImageCR(ctx, client, createOrUpdate, hcluster, controlPlaneNamespace)
+	if err != nil {
+		return nil, err
+	}
+
 	return openStackCluster, nil
 }
 
+func reconcileOpenStackImageCR(ctx context.Context, client client.Client, createOrUpdate upsert.CreateOrUpdateFN, hcluster *hyperv1.HostedCluster, controlPlaneNamespace string) error {
+	openStackImage := orc.Image{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "rhcos" + hcluster.Name,
+			Namespace: controlPlaneNamespace,
+		},
+		Spec: orc.ImageSpec{},
+	}
+
+	if _, err := createOrUpdate(ctx, client, &openStackImage, func() error {
```

Comment:
TODO: check what createOrUpdate() does. Given that we have applyconfigs it's trivial for us to SSA these, so there's really no need for a function that actually does create or update.

Reply:
From the doc:

```
// CreateOrUpdate is a copy of controllerutil.CreateOrUpdate with
// an important difference: It copies a number of fields from the object
// on the server to the mutated object if unset in the latter. This
// avoids unnecessary updates when our code sets a whole struct that
// has fields that get defaulted by the server.
```